sycl/source/detail/os_util.cpp

```diff
@@ -196,7 +196,7 @@ std::string OSUtil::getCurrentDSODir() {
   char Path[MAX_PATH];
   Path[0] = '\0';
   Path[sizeof(Path) - 1] = '\0';
-  auto Handle = getOSModuleHandle(&getCurrentDSODir);
+  auto Handle = getOSModuleHandle(reinterpret_cast<void *>(&getCurrentDSODir));
```

Comment:
This is to avoid this:
error: implicit conversion between pointer-to-function and pointer-to-object is a Microsoft extension [-Werror,-Wmicrosoft-cast]